velox/core/QueryCtx.cpp

```diff
@@ -79,4 +79,15 @@ QueryCtx::QueryCtx(
   return fmt::format("query.{}.{}", queryId.c_str(), seqNum++);
 }
 
+void QueryCtx::updateSpilledBytesAndCheckLimit(uint64_t bytes) {
+  const auto numSpilledBytes = numSpilledBytes_.fetch_add(bytes) + bytes;
+  if (queryConfig_.maxSpillBytes() > 0) {
+    VELOX_USER_CHECK_LE(
+        numSpilledBytes,
+        queryConfig_.maxSpillBytes(),
+        "Query exceeded per-query local spill limit of {}.",
+        succinctBytes(queryConfig_.maxSpillBytes()));
```

Comment:
Consider to add a histogram type of metric to report the spilled bytes on query ctx destruction? So we can see the the query node level spilled bytes distribution over the time in production? Thanks!